From b7e6f31df859191d42b9dde93b5d7e1c2a12f00f Mon Sep 17 00:00:00 2001
From: Jonathon Jongsma <jjongsma@redhat.com>
Date: Wed, 29 Nov 2017 10:41:54 -0600
Subject: [PATCH 47/55] Avoid leaking udscs file descriptor

This is a  finding from Coverity. In practice it has very little impact
since returning NULL from this function results in the process exiting
with an error.

Signed-off-by: Jonathon Jongsma <jjongsma@redhat.com>
Acked-by: Frediano Ziglio <fziglio@redhat.com>
---
 src/udscs.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/udscs.c b/src/udscs.c
index 64f0307..6e73015 100644
--- a/src/udscs.c
+++ b/src/udscs.c
@@ -450,12 +450,14 @@ struct udscs_server *udscs_create_server(const char *socketname,
     c = bind(fd, (struct sockaddr *)&address, sizeof(address));
     if (c != 0) {
         syslog(LOG_ERR, "bind %s: %m", socketname);
+        close(fd);
         return NULL;
     }
 
     c = listen(fd, 5);
     if (c != 0) {
         syslog(LOG_ERR, "listen: %m");
+        close(fd);
         return NULL;
     }
 
-- 
2.16.1

